x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsActionTests.java

```diff
@@ -0,0 +1,602 @@
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.http.MockRequest;
+import org.elasticsearch.test.http.MockResponse;
+import org.elasticsearch.test.http.MockWebServer;
+import org.elasticsearch.xpack.core.XPackSettings;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsAction;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsRequest;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsResponse;
+import org.elasticsearch.xpack.core.monitoring.exporter.MonitoringTemplateUtils;
+import org.elasticsearch.xpack.core.watcher.transport.actions.get.GetWatchAction;
+import org.elasticsearch.xpack.core.watcher.transport.actions.get.GetWatchRequest;
+import org.elasticsearch.xpack.monitoring.LocalStateMonitoring;
+import org.elasticsearch.xpack.monitoring.Monitoring;
+import org.elasticsearch.xpack.monitoring.MonitoringService;
+import org.elasticsearch.xpack.monitoring.exporter.ClusterAlertsUtil;
+import org.elasticsearch.xpack.monitoring.exporter.http.HttpExporter;
+import org.elasticsearch.xpack.monitoring.exporter.local.LocalExporter;
+import org.elasticsearch.xpack.monitoring.test.MockClusterAlertScriptEngine;
+import org.elasticsearch.xpack.monitoring.test.MockIngestPlugin;
+import org.elasticsearch.xpack.monitoring.test.MonitoringIntegTestCase;
+import org.junit.After;
+import org.junit.Before;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.xpack.monitoring.exporter.http.ClusterAlertHttpResource.CLUSTER_ALERT_VERSION_PARAMETERS;
+import static org.elasticsearch.xpack.monitoring.exporter.http.WatcherExistsHttpResource.WATCHER_CHECK_PARAMETERS;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.startsWith;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 3)
+public class TransportMonitoringMigrateAlertsActionTests extends MonitoringIntegTestCase {
+
+    private MockWebServer webServer;
+
+    private MockWebServer createMockWebServer() throws IOException {
+        MockWebServer server = new MockWebServer();
+        server.start();
+        return server;
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> transportClientPlugins() {
+        return Arrays.asList(LocalStateMonitoring.class, MockClusterAlertScriptEngine.TestPlugin.class,
+            MockIngestPlugin.class, CommonAnalysisPlugin.class);
+    }
+
+    @Override
+    protected Settings transportClientSettings() {
+        return Settings.builder().put(super.transportClientSettings())
+            .put(XPackSettings.SECURITY_ENABLED.getKey(), false)
+            .put(XPackSettings.WATCHER_ENABLED.getKey(), false)
+            .build();
+    }
```

Comment:
These were added:
1. To disable security in the tests. The tests were failing because security expects the Transport client type to be a secure one, but the tests install the mock-nio client.
2. To mirror the test plugins here in `transportClientPlugins` so that the client in the test has the watcher and monitoring actions available on it.

Reply:
are `MockIngestPlugin.class, CommonAnalysisPlugin.class` needed ?

Reply:
Nope! I took them off, along with `MockClusterAlertScriptEngine.TestPlugin.class`